ads/google/adsense.js

```diff
@@ -53,3 +60,55 @@ export function adsense(global, data) {
   global.document.getElementById('c').appendChild(i);
   (global.adsbygoogle = global.adsbygoogle || []).push({});
 }
+
+
+/**
+ * Make the ad iframe, with src=<the ad request>
+ * This makes the request directly, rather than using adsbygoogle.js.
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+function adsenseDirectRequest(global, data) {
+  checkData(data, ['adClient', 'adSlot', 'adHost', 'adtest', 'tagOrigin']);
+  if (global.context.clientId) {
+    // Read by GPT for GA/GPT integration.
+    global.gaGlobal = {
+      vid: global.context.clientId,
+      hid: global.context.pageViewId,
+    };
+  }
+
+  const adsenseInfo = getAdsenseInfo(global.context.master);
+  const slotNumber = adsenseInfo.nextSlotNumber();
+  makeAdsenseAd(global, data, slotNumber, global.context.initialIntersection);
+}
+
+/**
+ * Make an ad request.
+ * @param {!Window} global
+ * @param {!Object} data
+ * @param {number} slotNumber
+ * @param {!IntersectionObserverEntry} change
+ * @return {string}
+ */
+function makeAdsenseAd(global, data, slotNumber, change) {
+  const iframe = global.document.createElement('iframe');
+  const id = `google_ads_frame${slotNumber}`;
+  iframe.name = id;
```

Comment:
We had some very weird issues with browsers caching iframes with the same `name` (https://github.com/ampproject/amphtml/pull/2955). Should we add the url/src to this?